File: src/main/java/kitchenpos/order/tobe/eatinorder/domain/EatInOrderLineItem.java
package kitchenpos.order.tobe.eatinorder.domain;

import jakarta.persistence.*;

import java.math.BigDecimal;
import java.util.Objects;
import java.util.UUID;

@Table(name = "eat_in_order_line_item")
@Entity
public class EatInOrderLineItem {
    @Column(name = "seq")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    private Long seq;

    @Column(name = "menu_id", columnDefinition = "binary(16)")
    private UUID menuId;

    @Column(name = "quantity", nullable = false)
    private long quantity;

    @Embedded
    private EatInOrderPrice price;

    protected EatInOrderLineItem() {
    }

    public EatInOrderLineItem(UUID menuId, long quantity, BigDecimal price) {
        this.menuId = menuId;
        this.quantity = quantity;
        this.price = new EatInOrderPrice(price);
    }

    public UUID getMenuId() {
        return menuId;
    }

    public long getQuantity() {
        return quantity;
    }

    public BigDecimal getPrice() {
        return price.getPrice();
    }

    public Long getSeq() {
        return seq;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof EatInOrderLineItem orderLineItems)) {
            return false;
        }

        return this.getSeq() != null && Objects.equals(this.getSeq(), orderLineItems.getSeq());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getSeq());
    }
}

File: src/main/kotlin/kitchenpos/order/tobe/eatinorder/domain/EatInOrderLineItem.kt
package kitchenpos.order.tobe.eatinorder.domain

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import java.util.*
import kitchenpos.menu.tobe.domain.MenuDisplay
import kitchenpos.order.tobe.common.OrderMenuInfo

@Table(name = "order_line_item")
@Entity(name = "TobeOrderLineItem")
class EatInOrderLineItem(
    @Column(name = "seq")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    var seq: Long,

    @Column(name = "quantity", nullable = false)
    var quantity: Long,

    @Column(name = "menu_id")
    var menuId: UUID,
) {
    companion object {
        fun create(
            seq: Long,
            menuInfo: OrderMenuInfo,
            quantity: Long,
        ): EatInOrderLineItem {
            check(menuInfo.menuDisplay == MenuDisplay.DISPLAYED) { "노출한 메뉴만 주문할 수 있습니다." }
            return EatInOrderLineItem(
                seq = seq,
                quantity = quantity,
                menuId = menuInfo.menuId,
            )
        }
    }
}
